### PR TITLE
Add `published` output

### LINKED ISSUE #664: Output: list of published crates
Please add outputs to action: 
- list of published crates with versions
- list of not published crates with versions

It could be very useful for making release notes for workspaces.
Thanks! ❤️

### PR DESCRIPTION
**Pull request by boozook**
This is first time I touched ts, so please ensure all is right.

Partially fixes #664.

### CODE REVIEW

File: action.yml [hunk 1]
```diff
@@ -32,6 +32,9 @@ inputs:
   ignore-unpublished-changes:
     description: 'Exit the workflow gracefully if package does not have a new version to publish'
     default: 'false'
+outputs:
+  published:
+    description: 'JSON formatted string with published crates in object format { name => version }'
```
Comment (katyo) [inline action.yml:37]: Please describe it in README.

**Review by katyo — CHANGES_REQUESTED** [review]

**Comment by katyo** [comment]
@boozook Also I would like to see some tests.

File: src/main.ts [hunk 2]
```diff
@@ -123,11 +126,13 @@ async function run(): Promise<void> {
                     await exec('cargo', ['update', '--dry-run'], exec_opts)
                     info(`Package '${package_name}' published successfully`)
                 }
+                published[package_name] = package_info.version
```
[the remark this answers is not in the document]
Comment (boozook) [reply src/main.ts:129]: I purposely made it that way.
This way we always get a list of packages that:
- published
- would have been published if it weren't for dry-run.

My main resonses for this are:
1. The user can use this in combination with "dry-run"
2. If dry-run = true - the user knows this for sure, they set it.

What do you think about that? Am I really wrong?

File: action.yml (hunk 1 above)
Comment (boozook) [reply action.yml:37]: I will soon. Thank for review.

**Comment by boozook** [comment]
> @boozook Also I would like to see some tests.

Tests in ts is more complex thing for me. More complex than two simple lines of code. I'll try to beat it 🤪.

**Comment by boozook** [comment]
@katyo, I've looked at the tests and I have no idea how to do this, 
because it's not a unit test but a full system test to check the outgoing data. 
I really don't want to do that 🥺

File: action.yml (hunk 1 above)
Comment (boozook) [reply action.yml:37]: Probably solved in [that commit](https://github.com/katyo/publish-crates/pull/666/commits/555baf3ec4a5aff784dcf42dc5e78ebe51c4dc47).

**Comment by boozook** [comment]
I was thinking of a better implementation and I think it would be better to return an array with objects instead of just an object.
Like this:
```
[ { name: "name", version: "..." }, ... ]
```
This way it will be easier to get data out of it without additional conversions and extra steps.
Using [object filters](https://docs.github.com/en/actions/learn-github-actions/expressions#example-of-join) for [example](https://docs.github.com/en/actions/learn-github-actions/expressions#example-of-join).

I'll do it soon here.

**Review by katyo — APPROVED** [review]
LGTM

**Comment by katyo** [comment]
> @katyo, I've looked at the tests and I have no idea how to do this, because it's not a unit test but a full system test to check the outgoing data. I really don't want to do that 🥺

Ok. Changes looks quite simple, I think the tests hasn't so important in such case.

**Comment by katyo** [comment]
@boozook Please fix linting issues and I will merge this pr.

**Comment by boozook** [comment]
We've got couple listing errors.
My mistake is `error  'published' is never reassigned. Use 'const' instead`, will be fixed in a moment.
Others - I didn't touched that.

**Comment by boozook** [comment]
@katyo Thanks! Done.

**Comment by katyo** [comment]
@boozook Also please don't forget rebase and apply fixup commits. I haven't like to see it while merging.

**Comment by boozook** [comment]
@katyo, perhaps you could make a release?

**Comment by katyo** [comment]
@boozook v2 releases automatically when master branch updated.
See `dist/v2`.

**Pull request merged**